### PR TITLE
ISPN-8169 DirectoryImplementerTests.testConfigureAsyncDeletes random …

### PR DESCRIPTION
**Pull request by gustavonalle**
…failures

https://issues.jboss.org/browse/ISPN-8169

### CODE REVIEW

**Comment by gustavonalle** [comment]
Looks like CI is not picking this one

**Comment by anistor** [comment]
I noticed that. I'm running this locally in a loop to see how airtight it is.

**Comment by gustavonalle** [comment]
Even in master I can't replicated the failure in my local environment, even constraining severely the CPU to make it as slow as possible. There must be some serious hiccups happening on CI every now and then causing wild thread scheduling discrepancies

**Comment by anistor** [comment]
I think the initial version of the test may be wrong because it relied on the invocation of the ```afterExecute``` method of the threadpool, but the threadpool is shut down right after the second commit to the indexwriter. ```shutdown``` method guarantees that the previously submitted tasks are still executed but it does not say if the other 'contracts' of the threadpool will be honoured, like the execution of the ```afterExecute``` method, so we may have been relying on the wrong hook and the ```deletedSegments``` set may have not been updated correctly after the execution of the delete task. 

Anyway, I like the new version of this test. It's so much clearer than the previous.

I'm still running it trying to break it, but it does not seem to fail, so I'll merge it soon.

File: lucene/lucene-directory/src/test/java/org/infinispan/lucene/impl/DirectoryImplementerTests.java [hunk 1]
```diff
@@ -102,67 +101,43 @@ public void testGetIndexNameAndToString() throws IOException {
...
-      }
-
-      private String extractSegmentName(String fileName) {
-         if (!fileName.startsWith("_")) {
-            return null;
-         } else {
-            return fileName.substring(1, fileName.indexOf('.'));
-         }
-      }
+         verify(executor).execute(any(Runnable.class));
```
Comment (anistor) [inline lucene/lucene-directory/src/test/java/org/infinispan/lucene/impl/DirectoryImplementerTests.java]: Here I'd be very picky, and say ```verify(executor).execute(any(DirectoryLucene.DeleteTask.class))```

**Comment by gustavonalle** [comment]
Updated @anistor

**Comment by anistor** [comment]
Integrated in master. Thanks @gustavonalle !

**Comment by anistor** [comment]
Backported also to 8.2.x and 9.0.x.

**Pull request closed without merging**